typing/typecore.ml

```diff
@@ -845,6 +862,40 @@ let wrap_disambiguate msg ty f x =
   | Wrong_name_disambiguation (env, wrong_name) ->
     raise (Error (wrong_name.name.loc, env, Wrong_name (msg, ty, wrong_name)))
 
+(* Use a dummy instance of [NameChoice] to disambiguate between [float array]
+   and [floatarray]. Note that an actual ambiguous choice is not possible, as we
+   default to ['a array] in case of doubt. *)
+
+module FloatArray = NameChoice(struct
+    type t = bool
```

Comment:
I think this type should be expanded to be more like `label_description`. It should include the type of the array elements and the type of the resulting array as type expressions, alongside whether or not it is a float array. It would also be good to use a variant rather than a bool for that part.

This will allow you to implement `get_type` correctly rather than using `Predef.type_unit`. It will also allow you to use these types directly in `Ppat_array` and `Pexp_array` rather than computing them as `ty_elt` and `to_unify`.